core/src/main/java/io/questdb/griffin/engine/functions/math/GreatestNumericFunctionFactory.java

```diff
@@ -150,13 +159,12 @@ public ObjList<Function> getArgs() {
 
         @Override
         public double getDouble(Record rec) {
-            double value = Double.MIN_VALUE;
-            for (int i = 0, n = args.size(); i < n; i++) {
+            double value = args.getQuick(0).getDouble(rec);
+            for (int i = 1; i < n; i++) {
                 final double v = args.getQuick(i).getDouble(rec);
-                if (Numbers.isNull(v)) {
-                    return Double.NaN;
+                if (!Numbers.isNull(v)) {
+                    value = Math.max(value, v);
```

Comment:
The loop is incorrect in case when the first argument returns `NaN` - in that case the suggested function will return `NaN` even if all other arguments are non-`NaN`.

Reply:
I've added a few tests to verify the behavior around `null`s in 56c823dcf2f194a0908705bcb0f1d46cfb3d95af.